.github/workflows/ci.yml

```diff
@@ -15,7 +15,7 @@ env:
   RUST_BACKTRACE: 1
   # Change to specific Rust release to pin
   rust_stable: stable
-  rust_nightly: nightly-2023-10-21
+  rust_nightly: nightly-2024-05-05
```

Comment:
Ugh, upgrading the lint triggers an error that I don't know how to deal with.
```
error: data-layout for target `i686-unknown-linux-gnu-17863458210346918553`, `e-m:e-p:32:32-p270:32:32-p271:32:32-p272:64:64-f64:32:64-f80:32-n8:16:32-S128`, differs from LLVM target's `i686-unknown-linux-gnu` default layout, `e-m:e-p:32:32-p270:32:32-p271:32:32-p272:64:64-i128:128-f64:32:64-f80:32-n8:16:32-S128`
```
This is because of this CI step:

https://github.com/tokio-rs/tokio/blob/75e5b3d96d126b52a510ad19e79cf1e4445770e0/.github/workflows/ci.yml#L595-L625

Which tests Tokio using [this custom target](https://github.com/tokio-rs/tokio/blob/master/target-specs/i686-unknown-linux-gnu.json) to ensure that we work correctly when `AtomicU64` is missing.

cc @taiki-e any ideas?

Reply:
This is related to https://blog.rust-lang.org/2024/03/30/i128-layout-update.html

You should update the data-layout in your custom target:
https://github.com/tokio-rs/tokio/blob/75e5b3d96d126b52a510ad19e79cf1e4445770e0/target-specs/i686-unknown-linux-gnu.json#L5

Reply:
> cc @taiki-e any ideas?

The following command regenerates target-spec needed for this test.

```sh
rustc +nightly -Z unstable-options --print target-spec-json --target i686-unknown-linux-gnu | grep -v 'is-builtin' | sed 's/"max-atomic-width": 64/"max-atomic-width": 32/' > target-specs/i686-unknown-linux-gnu.json
```

(related: https://github.com/tokio-rs/tokio/issues/5187#issuecomment-1323280893)

Reply:
Thank you.